File: Nov2020/17Nov/IHateEvenSubarrays.java
/*
Problem Name: I hate Even Subarrays
Problem Link: https://www.hackerearth.com/practice/data-structures/stacks/basics-of-stacks/practice-problems/algorithm/i-hate-even-subarrays/
*/
import java.util.*;
class IHateEvenSubarrays {
    public static void main(String args[] ) throws Exception {
        Scanner in = new Scanner(System.in);
        int t = in.nextInt();
        for(;t>0;t--){
            String s = in.next();
            Stack<Character> st = new Stack<Character>();
            for(char c: s.toCharArray()){
                if(st.isEmpty())
                    st.push(c);
                else if(st.peek() == c)
                    st.pop();
                else
                    st.push(c);
            }
            if(st.isEmpty())
                System.out.println("KHALI");
            else{
                s = "";
                while(!st.isEmpty())
                    s = st.pop() + s;
                System.out.println(s);
            }
        }
    }
}

File: Nov2020/17Nov/IHateEvenSubarrays.cpp
/*
Problem Name: I hate Even Subarrays
Problem Link: https://www.hackerearth.com/practice/data-structures/stacks/basics-of-stacks/practice-problems/algorithm/i-hate-even-subarrays/
*/

#include<bits/stdc++.h>
using namespace std;
int main(){
    ios::sync_with_stdio(0);    
    cin.tie(0);             
    int t;
    cin>>t;
    while(t--){
        string s;
        int n,i=0;
        cin>>s;
        n=s.length();
        while(i<n){
            i++;
            if(i>0 && s[i]==s[i-1]){
                s.erase(i-1,2);
                i=i-2;
            }
        }
        if(s.length()>0)    cout<<s<<endl;
        else    cout<<"KHALI"<<endl;
    }
    return 0;
}
